plugin.py

```diff
@@ -0,0 +1,119 @@
+import sublime
+
+from LSP.plugin import AbstractPlugin, register_plugin, unregister_plugin
+from LSP.plugin.core.typing import Any, Optional
+
+import os
+import shutil
+import urllib.request
+
+
+USER_AGENT = 'Sublime Text LSP'
+
+TAG = '2022-08-27'
+
+MARKSMAN_RELEASES_BASE = 'https://github.com/artempyanykh/marksman/releases/download/{tag}/{platform}'
+MARKSMAN_FILENAME = 'marksman'
+
+
+def plat() -> Optional[str]:
+    '''
+    Returns the user friendly platform version that
+    sublime is running on.
+    '''
+    if sublime.platform() == 'osx':
+        return 'marksman-macos'
+    elif sublime.platform() == 'windows':
+        return 'marksman.exe'
+    elif sublime.platform() == 'linux':
+        return 'marksman-linux'
+    else:
+        return None
+
+
+class Marksman(AbstractPlugin):
+    '''
+    Marksman is an AbstractPlugin implementation that provides
+    the required functions to act as a helper package for the
+    Markdown Language Server (marksman)
+    '''
+
+    @classmethod
+    def name(cls) -> str:
+        return "marksman"
+
+    @classmethod
+    def basedir(cls) -> str:
+        return os.path.join(cls.storage_path(), __package__)
+
+    @classmethod
+    def server_version(cls) -> str:
+        return TAG
+
+    @classmethod
+    def current_server_version(cls) -> Optional[str]:
+        try:
+            with open(os.path.join(cls.basedir(), "VERSION"), "r") as fp:
+                return fp.read()
+        except:
+            return None
+
+    @classmethod
+    def _get_marksman_ls_path(cls) -> str:
+        binary = 'marksman.exe' if sublime.platform() == 'windows' else 'marksman'
+        command = get_setting(
+            'command', [os.path.join(cls.basedir(), 'bin', binary)]
+        )
+        marksman_ls_binary = command[0].replace('${storage_path}', cls.storage_path())
```

Comment:
This code is so messed up...
- It only processes the first argument of `command` which happens to be a full path in this server but in practice it can be an array of arguments in which case all further arguments would be dropped...
- If user overrides the `command`, which I guess is the point of handling it this way, then the code will still try to install the server in package storage which doesn't quite make sense.